paasta_tools/setup_istio_mesh.py

```diff
@@ -67,21 +69,35 @@ def parse_args() -> argparse.Namespace:
         help="Update or create up to this number of service instances. Default is 0 (no limit).",
     )
     parser.add_argument(
-        "-f",
-        "--file-path",
-        dest="file_path",
-        default="/nail/etc/services/services.json",
+        "-d",
+        "--soa-dir",
+        dest="soa_dir",
+        default=DEFAULT_SOA_DIR,
         metavar="LIMIT",
         type=str,
-        help="services file to use for smartstack namespace. Default file is /nail/etc/services/services.json",
+        help=f"Directory with service declarations. Default is {DEFAULT_SOA_DIR}",
     )
     args = parser.parse_args()
     return args
 
 
-def load_smartstack_namespaces(file_path: str) -> Mapping:
-    with open(file_path) as service_file:
-        return json.load(service_file)
+def load_smartstack_namespaces(soa_dir: str = DEFAULT_SOA_DIR) -> Mapping:
+    namespaces = {}
+
+    _, dirs, _ = next(os.walk(soa_dir))
```

Comment:
Not important for now, but in future maybe want to check how long it takes us iterate over the whole yelpsoa-config since it's quite big, but for now this seems like a good enough choice.

Reply:
yeah, I'm also concerned about this being slow (can you add a timer or something to get some understanding of how long this takes?)- can't we read these concurrently?

Reply:
this line doesn't iterate entire tree, only first level of hierarchy. takes 150 milliseconds. it only opens single smartstack.yaml file per service and most often that file will already be in cache.